{"ygPerms":{"resourceCapabilityList":[{"resourceType":"GROUP","capabilities":[{"name":"READ"},{"name":"JOIN"}]},{"resourceType":"PHOTO","capabilities":[]},{"resourceType":"FILE","capabilities":[]},{"resourceType":"MEMBER","capabilities":[]},{"resourceType":"LINK","capabilities":[]},{"resourceType":"CALENDAR","capabilities":[]},{"resourceType":"DATABASE","capabilities":[]},{"resourceType":"POLL","capabilities":[]},{"resourceType":"MESSAGE","capabilities":[{"name":"READ"}]},{"resourceType":"PENDING_MESSAGE","capabilities":[]},{"resourceType":"ATTACHMENTS","capabilities":[{"name":"READ"}]},{"resourceType":"PHOTOMATIC_ALBUMS","capabilities":[]},{"resourceType":"MEMBERSHIP_TYPE","capabilities":[]},{"resourceType":"POST","capabilities":[{"name":"READ"}]},{"resourceType":"PIN","capabilities":[]}],"groupUrl":"groups.yahoo.com","intlCode":"us"},"comscore":"pageview_candidate","ygData":{"userId":137285340,"authorName":"Gordon Mohr","from":"Gordon Mohr &lt;gojomo@...&gt;","profile":"gojomo","replyTo":"LIST","senderId":"N9e5j7xzCX5VipyZU0pdShW3trtYdNAbobXvAExVCqbSQAZEb2ygKkK-9OA6IJsuAYC8NAirCUEZ-c2oqc2BKrJedTXabV0","spamInfo":{"isSpam":false,"reason":"2"},"subject":"Re: cannot recover from checkpoint","postDate":"1327024767","msgId":7541,"canDelete":false,"contentTrasformed":false,"systemMessage":false,"headers":{"messageIdInHeader":"PDRGMThDQTdGLjkwNzAyMDBAYXJjaGl2ZS5vcmc+","inReplyToHeader":"PDRGMTcxQ0UyLjIwOTAwMDZAY3MuY211LmVkdT4=","referencesHeader":"PDRGMTcxQ0UyLjIwOTAwMDZAY3MuY211LmVkdT4="},"prevInTopic":7536,"nextInTopic":7543,"prevInTime":7540,"nextInTime":7542,"topicId":7532,"numMessagesInTopic":6,"msgSnippet":"I m not sure what s causing the errors about the ??text/javascript keys. Perhaps an encoding problem which has had the effect of creating bad JSON. Manually","rawEmail":"Return-Path: &lt;gojomo@...&gt;\r\nX-Sender: gojomo@...\r\nX-Apparently-To: archive-crawler@yahoogroups.com\r\nX-Received: (qmail 66612 invoked from network); 20 Jan 2012 01:59:30 -0000\r\nX-Received: from unknown (98.137.34.45)\n  by m13.grp.sp2.yahoo.com with QMQP; 20 Jan 2012 01:59:30 -0000\r\nX-Received: from unknown (HELO relay00.pair.com) (209.68.5.9)\n  by mta2.grp.sp2.yahoo.com with SMTP; 20 Jan 2012 01:59:29 -0000\r\nX-Received: (qmail 31278 invoked by uid 0); 20 Jan 2012 01:59:28 -0000\r\nX-Received: from 76.218.213.38 (HELO silverbook.local) (76.218.213.38)\n  by relay00.pair.com with SMTP; 20 Jan 2012 01:59:28 -0000\r\nX-pair-Authenticated: 76.218.213.38\r\nMessage-ID: &lt;4F18CA7F.9070200@...&gt;\r\nDate: Thu, 19 Jan 2012 17:59:27 -0800\r\nUser-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.7; rv:9.0) Gecko/20111222 Thunderbird/9.0.1\r\nMIME-Version: 1.0\r\nTo: David Pane &lt;dpane@...&gt;, archive-crawler@yahoogroups.com\r\nCc: Noah Levitt &lt;nlevitt@...&gt;\r\nReferences: &lt;4F171CE2.2090006@...&gt;\r\nIn-Reply-To: &lt;4F171CE2.2090006@...&gt;\r\nContent-Type: text/plain; charset=ISO-8859-1; format=flowed\r\nContent-Transfer-Encoding: 7bit\r\nX-eGroups-Msg-Info: 2:2:2:0:1\r\nFrom: Gordon Mohr &lt;gojomo@...&gt;\r\nSubject: Re: cannot recover from checkpoint\r\nX-Yahoo-Group-Post: member; u=137285340; y=U5o0-Ly3C7WTFWYJDaLO8_JeJBdcNVID_99Fco9RZW93\r\nX-Yahoo-Profile: gojomo\r\n\r\nI&#39;m not sure what&#39;s causing the errors about the &quot;??text/javascript&quot; \nkeys. Perhaps an encoding problem which has had the effect of creating \nbad JSON. Manually deleting the &#39;duplicate key&#39; from the \n&#39;statisticsTracker&#39; file in the checkpoint directory would, at the loss \nof some statistics, probably workaround that error.\n\nThat may not be the most severe problem with the checkpoint(s), though. \nThere are quite a number of SEVERE errors like:\n\n2011-12-26 08:02:35.522 SEVERE thread-15 \norg.archive.bdb.BdbModule.doCheckpoint() unable to create required \ncheckpoint link \n/scratch/heritrix-3.1.1/jobs/2011-12-23/state/cp00074-20111226080024/00003737.jdb,9999433\n...\n2012-01-07 15:45:08.948 SEVERE thread-1217 \norg.archive.bdb.BdbModule.doCheckpoint() unable to create required \ncheckpoint link \n/scratch/heritrix-3.1.1/jobs/2011-12-23/state/cp00046-20120107154231/00061748.jdb,9999989\n...\n2012-01-16 23:19:48.606 SEVERE thread-1217 \norg.archive.bdb.BdbModule.doCheckpoint() unable to create required \ncheckpoint link \n/scratch/heritrix-3.1.1/jobs/2011-12-23/state/cp00268-20120116231616/0009dbed.jdb,9998880\n\nSince even one of these indicates a necessary file isn&#39;t being saved \n(linked) into the checkpoint directory, it could be fatal for that \naffected checkpoint. (I think the fastest way to tell, after the fact, \nthat such a checkpoint is missing a file is if the jdbfiles.manifest in \nthe state-dir checkpoint-dir lists a filename that isn&#39;t present. If a \nfilename of that same name was still present in the main state \ndirectory, it would usually be substitutable.)\n\nTo find out what&#39;s happening, we&#39;d need to take a closer look at the \nmachine around the time they happen. Are there other disk problems, \nevident in Heritrix or system logs? Is this a transient problem that (on \nnext checkpoint attempt) usually succeeds for the exact same files? (If \nso maybe we need the normal checkpoint process to be in a retry loop.)\n\n- Gordon\n\nOn 1/18/12 11:26 AM, David Pane wrote:\n&gt; This morning I shut down all 5 instances. 4 recovered from the last\n&gt; checkpoint and are crawling. They are all running on less than\n&gt; configured threads due to the server errors I&#39;ve seen before.\n&gt;\n&gt; http://tech.groups.yahoo.com/group/archive-crawler/message/7493\n&gt;\n&gt; It is more severe than it was the last time.\n&gt;\n&gt; Instance #0 has 40 less - running with 1160.\n&gt; Instance #2 has 234 less - running with 234.\n&gt; Instance #3 has 111 less - running with 111.\n&gt; Instance #4 has 26 less - running with 26.\n&gt;\n&gt; -------------\n&gt; Instance #1:\n&gt; -------------\n&gt; When attempting to recover from &quot;any&quot; recent checkpoint, I get the\n&gt; following errors (see below) and the recover fails. Although this\n&gt; instance did not give me problems before, there were multiple SEVERE\n&gt; errors in the logs previous to me stopping the crawl. see attached file\n&gt; - I removed all INFO lines from the logs - sorry for the large\n&gt; attachment....\n&gt;\n&gt; I found that the mime-report has two entries for &quot;??text/javascript&quot;\n&gt;\n&gt;  &gt;&gt;&gt; grep javascript mimetype-report.txt.2012-01-17h19m05\n&gt; 726509 15900289997 application/x-javascript\n&gt; 277744 5991936365 application/javascript\n&gt; 127275 3432127316 text/javascript\n&gt; 561 146075 text/javascript+json\n&gt; 272 7486875 text/x-javascript\n&gt; 24 142128 javascript/text\n&gt; 15 155529 javascript\n&gt; 10 191371 ?text/javascript?\n&gt; 6 175283 javascript/x-javascript\n&gt; 4 80925 application/vnd.javascript\n&gt; 3 193869 type/javascript\n&gt; 1 74208 ??text/javascript\n&gt; 1 33306 txt/javascript\n&gt; 1 74208 ??text/javascript\n&gt;\n&gt; I am also ready to start a recover from the frontier.recover but I am\n&gt; hoping that there is a way around this problem.... any ideas?????\n&gt;\n&gt; --David\n&gt;\n&gt; Here is a sample of what is in the logs.\n&gt;\n&gt; Job.log:\n&gt;\n&gt; 2012-01-18T17:37:39.992Z SEVERE Failed to start bean\n&gt; &#39;statisticsTracker&#39;; nested exception is java.lang.RuntimeException:\n&gt; org.json.JSONException: Duplicate key &quot;??text/javascript&quot; (in thread\n&gt; &#39;2011-12-23 launchthread&#39;)\n&gt; 2012-01-18T17:37:39.992Z SEVERE Failed to start bean\n&gt; &#39;statisticsTracker&#39;; nested exception is java.lang.RuntimeException:\n&gt; org.json.JSONException: Duplicate key &quot;??text/javascript&quot;\n&gt; 2012-01-18T17:37:39.990Z INFO Job instance discarded\n&gt; 2012-01-18T17:04:00.791Z SEVERE unable to delete obstructing file\n&gt; /scratch/heritrix-3.1.1/jobs/2011-12-23/state/00088669.jdb (in thread\n&gt; &#39;2011-12-23 launchthread&#39;)\n&gt; 2012-01-18T17:03:55.033Z SEVERE unable to create required restore link\n&gt; /scratch/heritrix-3.1.1/jobs/2011-12-23/state/00081efd.jdb (in thread\n&gt; &#39;2011-12-23 launchthread&#39;)\n&gt; 2012-01-18T17:02:18.458Z INFO Job launched\n&gt; 2012-01-18T16:59:19.283Z INFO Job instantiated\n&gt;\n&gt;\n&gt;\n&gt; ---------------------------------------\n&gt; cat heritrix_out.log\n&gt;\n&gt; Wed Jan 18 11:59:05 EST 2012 Starting heritrix\n&gt; Linux compute-1-12.local 2.6.18-194.17.4.el5 #1 SMP Mon Oct 25 15:50:53\n&gt; EDT 2010 x86_64 GNU/Linux\n&gt; java version &quot;1.6.0_16&quot;\n&gt; Java(TM) SE Runtime Environment (build 1.6.0_16-b01)\n&gt; Java HotSpot(TM) 64-Bit Server VM (build 14.2-b01, mixed mode)\n&gt; JAVA_OPTS=-Xmx55000M\n&gt; core file size (blocks, -c) 0\n&gt; data seg size (kbytes, -d) unlimited\n&gt; scheduling priority (-e) 0\n&gt; file size (blocks, -f) unlimited\n&gt; pending signals (-i) 530432\n&gt; max locked memory (kbytes, -l) unlimited\n&gt; max memory size (kbytes, -m) unlimited\n&gt; open files (-n) 4096\n&gt; pipe size (512 bytes, -p) 8\n&gt; POSIX message queues (bytes, -q) 819200\n&gt; real-time priority (-r) 0\n&gt; stack size (kbytes, -s) unlimited\n&gt; cpu time (seconds, -t) unlimited\n&gt; max user processes (-u) 530432\n&gt; virtual memory (kbytes, -v) unlimited\n&gt; file locks (-x) unlimited\n&gt; Sun Microsystems Inc. Java(TM) SE Runtime Environment 1.6.0_16-b01\n&gt; Using ad-hoc HTTPS certificate with fingerprint...\n&gt; SHA1:7E:D5:A0:17:AD:CB:03:7D:4C:AD:FB:E7:04:41:AB:0B:1A:42:78:B3\n&gt; Verify in browser before accepting exception.\n&gt; 2012-01-18 16:59:05.737 WARNING thread-10\n&gt; org.archive.crawler.framework.Engine.findJobConfigs() invalid job\n&gt; directory: ./jobs/var where job expected from: ./jobs/var\n&gt; 2012-01-18 16:59:05.758 WARNING thread-10\n&gt; org.archive.crawler.framework.Engine.findJobConfigs() invalid job\n&gt; directory: ./jobs/heritrix_dir where job expected from: ./jobs/heritrix_dir\n&gt; 2012-01-18 16:59:05.892 INFO thread-10\n&gt; org.archive.crawler.framework.Engine.addJobDirectory() added crawl job:\n&gt; 2011-12-23\n&gt; 2012-01-18 16:59:05.893 WARNING thread-10\n&gt; org.archive.crawler.framework.Engine.findJobConfigs() invalid job\n&gt; directory: ./jobs/heritrix_out.logSAVE~ where job expected from:\n&gt; ./jobs/heritrix_out.logSAVE~\n&gt; 2012-01-18 16:59:05.893 WARNING thread-10\n&gt; org.archive.crawler.framework.Engine.findJobConfigs() invalid job\n&gt; directory: ./jobs/heritrix_out.logSAVE where job expected from:\n&gt; ./jobs/heritrix_out.logSAVE\n&gt; 2012-01-18 16:59:05.894 WARNING thread-10\n&gt; org.archive.crawler.framework.Engine.findJobConfigs() invalid job\n&gt; directory: ./jobs/heritrix_out.log20120118 where job expected from:\n&gt; ./jobs/heritrix_out.log20120118\n&gt; 2012-01-18 16:59:05.894 WARNING thread-10\n&gt; org.archive.crawler.framework.Engine.findJobConfigs() invalid job\n&gt; directory: ./jobs/heritrix_out.log20120118no-INFO where job expected\n&gt; from: ./jobs/heritrix_out.log20120118no-INFO\n&gt; 2012-01-18 16:59:06.04::INFO: Logging to STDERR via\n&gt; org.mortbay.log.StdErrLog\n&gt; 2012-01-18 16:59:06.05::INFO: jetty-6.1.14\n&gt; 2012-01-18 16:59:06.369::INFO: Started SslSocketConnector@10.1.1.249:8443\n&gt; engine listening at port 8443\n&gt; operator login set per file @...\n&gt; Heritrix version: 3.1.1-SNAPSHOT-20111222.225453\n&gt; 2012-01-18 16:59:17.201 INFO thread-11\n&gt; org.springframework.context.support.AbstractApplicationContext.prepareRefresh()\n&gt; Refreshing org.archive.spring.PathSharingContext@44ed904: startup date\n&gt; [Wed Jan 18 16:59:17 GMT 2012]; root of context hierarchy\n&gt; 2012-01-18 16:59:19.283 INFO thread-11\n&gt; org.archive.crawler.framework.CrawlJob.instantiateContainer() Job\n&gt; instantiated\n&gt; 2012-01-18 17:02:18.458 INFO thread-12\n&gt; org.archive.crawler.framework.CrawlJob.launch() Job launched\n&gt; 2012-01-18 17:02:19.997 INFO thread-13\n&gt; org.archive.spring.PathSharingContext.initLaunchId() launch id\n&gt; 20120118170219\n&gt; 2012-01-18 17:02:20.047 INFO thread-13\n&gt; org.springframework.context.support.DefaultLifecycleProcessor$LifecycleGroup.start()\n&gt; Starting beans in phase 0\n&gt; 2012-01-18 17:03:55.033 SEVERE thread-13\n&gt; org.archive.bdb.BdbModule.doRecover() unable to create required restore\n&gt; link /scratch/heritrix-3.1.1/jobs/2011-12-23/state/00081efd.jdb\n&gt; 2012-01-18 17:04:00.791 SEVERE thread-13\n&gt; org.archive.bdb.BdbModule.doRecover() unable to delete obstructing file\n&gt; /scratch/heritrix-3.1.1/jobs/2011-12-23/state/00088669.jdb\n&gt; 2012-01-18 17:34:12.297 INFO thread-13\n&gt; org.springframework.context.support.AbstractApplicationContext.doClose()\n&gt; Closing org.archive.spring.PathSharingContext@44ed904: startup date [Wed\n&gt; Jan 18 16:59:17 GMT 2012]; root of context hierarchy\n&gt; 2012-01-18 17:34:12.298 INFO thread-13\n&gt; org.springframework.context.support.DefaultLifecycleProcessor$LifecycleGroup.stop()\n&gt; Stopping beans in phase 0\n&gt; 2012-01-18 17:37:39.990 INFO thread-13\n&gt; org.archive.crawler.framework.CrawlJob.doTeardown() Job instance discarded\n&gt; 2012-01-18 17:37:39.992 SEVERE thread-13\n&gt; org.archive.crawler.framework.CrawlJob.beansException() Failed to start\n&gt; bean &#39;statisticsTracker&#39;; nested exception is\n&gt; java.lang.RuntimeException: org.json.JSONException: Duplicate key\n&gt; &quot;??text/javascript&quot;\n&gt; org.springframework.context.ApplicationContextException: Failed to start\n&gt; bean &#39;statisticsTracker&#39;; nested exception is\n&gt; java.lang.RuntimeException: org.json.JSONException: Duplicate key\n&gt; &quot;??text/javascript&quot;\n&gt; at\n&gt; org.springframework.context.support.DefaultLifecycleProcessor.doStart(DefaultLifecycleProcessor.java:169)\n&gt;\n&gt; at\n&gt; org.springframework.context.support.DefaultLifecycleProcessor.doStart(DefaultLifecycleProcessor.java:159)\n&gt;\n&gt; at\n&gt; org.springframework.context.support.DefaultLifecycleProcessor.doStart(DefaultLifecycleProcessor.java:159)\n&gt;\n&gt; at\n&gt; org.springframework.context.support.DefaultLifecycleProcessor.doStart(DefaultLifecycleProcessor.java:159)\n&gt;\n&gt; at\n&gt; org.springframework.context.support.DefaultLifecycleProcessor.access$1(DefaultLifecycleProcessor.java:154)\n&gt;\n&gt; at\n&gt; org.springframework.context.support.DefaultLifecycleProcessor$LifecycleGroup.start(DefaultLifecycleProcessor.java:335)\n&gt;\n&gt; at\n&gt; org.springframework.context.support.DefaultLifecycleProcessor.startBeans(DefaultLifecycleProcessor.java:143)\n&gt;\n&gt; at\n&gt; org.springframework.context.support.DefaultLifecycleProcessor.start(DefaultLifecycleProcessor.java:89)\n&gt;\n&gt; at\n&gt; org.springframework.context.support.AbstractApplicationContext.start(AbstractApplicationContext.java:1236)\n&gt;\n&gt; at org.archive.spring.PathSharingContext.start(PathSharingContext.java:112)\n&gt; at org.archive.crawler.framework.CrawlJob.startContext(CrawlJob.java:458)\n&gt; at org.archive.crawler.framework.CrawlJob$1.run(CrawlJob.java:432)\n&gt; Caused by: java.lang.RuntimeException: org.json.JSONException: Duplicate\n&gt; key &quot;??text/javascript&quot;\n&gt; at org.archive.checkpointing.Checkpoint.loadJson(Checkpoint.java:142)\n&gt; at\n&gt; org.archive.crawler.reporting.StatisticsTracker.start(StatisticsTracker.java:355)\n&gt;\n&gt; at\n&gt; org.springframework.context.support.DefaultLifecycleProcessor.doStart(DefaultLifecycleProcessor.java:166)\n&gt;\n&gt; ... 11 more\n&gt; Caused by: org.json.JSONException: Duplicate key &quot;??text/javascript&quot;\n&gt; at org.json.JSONObject.putOnce(JSONObject.java:1076)\n&gt; at org.json.JSONObject.&lt;init&gt;(JSONObject.java:205)\n&gt; at org.json.JSONTokener.nextValue(JSONTokener.java:344)\n&gt; at org.json.JSONObject.&lt;init&gt;(JSONObject.java:205)\n&gt; at org.json.JSONObject.&lt;init&gt;(JSONObject.java:402)\n&gt; at org.archive.checkpointing.Checkpoint.loadJson(Checkpoint.java:140)\n&gt; ... 13 more\n&gt; Exception in thread &quot;2011-12-23 launchthread&quot;\n&gt; org.springframework.beans.factory.BeanCreationException: Error creating\n&gt; bean with name &#39;configPathConfigurer&#39; defined in URL\n&gt; [file:/scratch/heritrix-3.1.1/jobs/2011-12-23/crawler-beans.cxml]:\n&gt; Initialization of bean failed; nested exception is\n&gt; java.lang.IllegalStateException: BeanFactory not initialized or already\n&gt; closed - call &#39;refresh&#39; before accessing beans via the ApplicationContext\n&gt; at\n&gt; org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:527)\n&gt;\n&gt; at\n&gt; org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:456)\n&gt;\n&gt; at\n&gt; org.springframework.beans.factory.support.AbstractBeanFactory$1.getObject(AbstractBeanFactory.java:291)\n&gt;\n&gt; at\n&gt; org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:222)\n&gt;\n&gt; at\n&gt; org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:288)\n&gt;\n&gt; at\n&gt; org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:194)\n&gt;\n&gt; at\n&gt; org.springframework.context.event.AbstractApplicationEventMulticaster.getApplicationListeners(AbstractApplicationEventMulticaster.java:148)\n&gt;\n&gt; at\n&gt; org.springframework.context.event.SimpleApplicationEventMulticaster.multicastEvent(SimpleApplicationEventMulticaster.java:86)\n&gt;\n&gt; at\n&gt; org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:303)\n&gt;\n&gt; at\n&gt; org.archive.crawler.framework.CrawlController.sendCrawlStateChangeEvent(CrawlController.java:326)\n&gt;\n&gt; at\n&gt; org.archive.crawler.framework.CrawlController.requestCrawlStart(CrawlController.java:347)\n&gt;\n&gt; at org.archive.crawler.framework.CrawlJob$1.run(CrawlJob.java:434)\n&gt; Caused by: java.lang.IllegalStateException: BeanFactory not initialized\n&gt; or already closed - call &#39;refresh&#39; before accessing beans via the\n&gt; ApplicationContext\n&gt; at\n&gt; org.springframework.context.support.AbstractRefreshableApplicationContext.getBeanFactory(AbstractRefreshableApplicationContext.java:171)\n&gt;\n&gt; at\n&gt; org.springframework.context.support.AbstractApplicationContext.containsBean(AbstractApplicationContext.java:1091)\n&gt;\n&gt; at\n&gt; org.springframework.context.support.AbstractApplicationContext$ApplicationListenerDetector.postProcessAfterInitialization(AbstractApplicationContext.java:1369)\n&gt;\n&gt; at\n&gt; org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.applyBeanPostProcessorsAfterInitialization(AbstractAutowireCapableBeanFactory.java:407)\n&gt;\n&gt; at\n&gt; org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1426)\n&gt;\n&gt; at\n&gt; org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:519)\n&gt;\n&gt; ... 11 more\n&gt;\n\n"}}